lib/configgin.rb

```diff
@@ -59,23 +60,43 @@ def generate_jobs(job_configs, templates)
 
   # Set the exported properties and their digests, and return the digests.
   def patch_job_metadata(jobs)
+    pod = kube_client.get_pod(@self_name, kube_namespace)
+
+    secret = Kubeclient::Resource.new
+    secret.metadata = {}
+    # Prefixing with pod.metadata.name is purely for human convenience/debugging.
+    secret.metadata.name = "#{pod.metadata.name}-#{pod.metadata.uid}"
```

Comment:
Would it be useful to also prefix the secret name with `configgin-` or something, to identify where it comes from / sort better?

FWIW, it looks like secret names can be up to 253 characters long.

Reply:
I don't want to prefix them because I want `k get secret scf:uaa` to do the right thing. That is the point why they are prefixed with the pod name at all.

We can add a suffix, but that doesn't help with the sorting, and the format of the `uid` makes it pretty clear which secrets are of this kind. Adding a suffix makes the output wider, making it more likely to wrap in the terminal, email, or chat programs, so I would prefer to keep it as is.

Let me know if you disagree strongly, and I'll add a suffix... 😄

Reply:
`k` actually uses a regex match, so `k get secret scf:uaa` would still do the right thing… But yeah I wouldn't want a suffix; either prefix or nothing.

Reply:
> But yeah I wouldn't want a suffix; either prefix or nothing.

Ok, then you get nothing! 😛 

Because the prefix would still break `k get secret scf:^router`, and you need the `^` anchor to disambiguate between `router` and `tcp-router`...